src/execution.js

```diff
@@ -24,24 +34,29 @@ import { ajax, post } from './xhr';
  * @return {Object} Structure with `headers` and `rawData` keys filled with values from execution.
  */
 export function getData(projectId, elements, executionConfiguration) {
-    // Create request and result structures
-    const request = {
-        execution: {
-            columns: elements
-        }
-    };
+    let request;
     const executedReport = {
         isLoaded: false
     };
 
-    // enrich configuration with supported properties such as
-    // where clause with query-like filters or execution context filters
-    const config = executionConfiguration || {};
-    ['filters', 'where', 'orderBy', 'definitions'].forEach(function assignProperties(property) {
-        if (config[property]) {
-            request.execution[property] = config[property];
-        }
-    });
+    if (isPlainObject(elements)) {
+        request = elements;
+    } else {
+        // Create request and result structures
+        request = {
+            execution: {
+                columns: elements
+            }
+        };
+        // enrich configuration with supported properties such as
+        // where clause with query-like filters or execution context filters
+        const config = executionConfiguration || {};
```

Comment:
this could be assign as executionConfiguration default parameter in method signature